pkg/cmd/release/create/create.go

```diff
@@ -74,88 +110,67 @@ func NewCmdCreate(f factory.Factory) *cobra.Command {
+	flags.StringVarP(&createFlags.PackageVersion.Value, createFlags.PackageVersion.Name, "", "", "Default version to use for all Packages")
+	flags.StringVarP(&createFlags.ReleaseNotes.Value, createFlags.ReleaseNotes.Name, "n", "", "Release notes to attach")
+	flags.StringVarP(&createFlags.Version.Value, createFlags.Version.Name, "v", "", "Override the Release Version")
+	flags.BoolVarP(&createFlags.IgnoreExisting.Value, createFlags.IgnoreExisting.Name, "x", false, "If a release with the same version exists, do nothing instead of failing.")
+	flags.BoolVarP(&createFlags.IgnoreChannelRules.Value, createFlags.IgnoreChannelRules.Name, "", false, "Allow creation of a release where channel rules would otherwise prevent it.")
+	flags.StringSliceVarP(&createFlags.PackageVersionSpec.Value, createFlags.PackageVersionSpec.Name, "", []string{}, "Version specification a specific packages.\nFormat as {package}:{version}, {step}:{version} or {package-ref-name}:{packageOrStep}:{version}\nYou may specify this multiple times")
 
 	// we want the help text to display in the above order, rather than alphabetical
 	flags.SortFlags = false
 
 	// flags aliases for compat with old .NET CLI
-	util.AddFlagAliasesString(flags, FlagGitRef, FlagAliasGitRefRef, FlagAliasGitRefLegacy)
-	util.AddFlagAliasesString(flags, FlagGitCommit, FlagAliasGitCommitLegacy)
-	util.AddFlagAliasesString(flags, FlagPackageVersion, FlagAliasDefaultPackageVersion, FlagAliasPackageVersionLegacy, FlagAliasDefaultPackageVersionLegacy)
-	util.AddFlagAliasesString(flags, FlagReleaseNotes, FlagAliasReleaseNotesLegacy)
-	util.AddFlagAliasesString(flags, FlagVersion, FlagAliasReleaseNumberLegacy)
-	util.AddFlagAliasesString(flags, FlagPackagePrerelease, FlagAliasPackagePrereleaseLegacy)
-	util.AddFlagAliasesBool(flags, FlagIgnoreExisting, FlagAliasIgnoreExistingLegacy)
-	util.AddFlagAliasesBool(flags, FlagIgnoreChannelRules, FlagAliasIgnoreChannelRulesLegacy)
+	flagAliases := make(map[string][]string, 10)
+	util.AddFlagAliasesString(flags, FlagGitRef, flagAliases, FlagAliasGitRefRef, FlagAliasGitRefLegacy)
+	util.AddFlagAliasesString(flags, FlagGitCommit, flagAliases, FlagAliasGitCommitLegacy)
+	util.AddFlagAliasesString(flags, FlagPackageVersion, flagAliases, FlagAliasDefaultPackageVersion, FlagAliasPackageVersionLegacy, FlagAliasDefaultPackageVersionLegacy)
+	util.AddFlagAliasesString(flags, FlagReleaseNotes, flagAliases, FlagAliasReleaseNotesLegacy)
+	util.AddFlagAliasesString(flags, FlagVersion, flagAliases, FlagAliasReleaseNumberLegacy)
+	util.AddFlagAliasesBool(flags, FlagIgnoreExisting, flagAliases, FlagAliasIgnoreExistingLegacy)
+	util.AddFlagAliasesBool(flags, FlagIgnoreChannelRules, flagAliases, FlagAliasIgnoreChannelRulesLegacy)
+
+	cmd.PreRunE = func(cmd *cobra.Command, args []string) error {
```

Comment:
I worked out a much nicer way of dealing with flag aliases. Could probably tighten it up a bit further, but I think this should do for now.

Logic might not be particularly clear so let me know if you'd like a run-through @domenicsim1 @jbristowe